PR: https://github.com/odoo/odoo/pull/

From: 7a9dd47ae573b26ff7a5c9460bb297646b8ce63c
From: Pierre Masereel
Date: 2017-08-04 13:06:19

Structural Changes: 5
Total Changes: 24

[REF] stock, mrp, mrp_repair: implement mrp and mrp_repair traceability

The traceability report didn't take yet in consideration the consumed
and produced lines generated by manufacturing orders and repair orders.

================================= pseudo patch: =================================

--- a/addons/mrp_repair/models/__init__.py
+++ b/addons/mrp_repair/models/__init__.py
@@ -2,3 +2,4 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import mrp_repair
+from . import stock_traceability

--- a/addons/mrp_repair/models/mrp_repair.py
+++ b/addons/mrp_repair/models/mrp_repair.py
@@ -12,14 +12,6 @@ class StockMove(models.Model):
     _inherit = 'stock.move'
 
     repair_id = fields.Many2one('mrp.repair')
-    consume_repair_id = fields.Many2one('mrp.repair')
-
-
-class StockMoveLine(models.Model):
-    _inherit = 'stock.move.line'
-
-    repair_id = fields.Many2one('mrp.repair', related='move_id.repair_id')
-    consume_repair_id = fields.Many2one('mrp.repair', related='move_id.consume_repair_id')
 
 
 class Repair(models.Model):
@@ -442,7 +434,7 @@ class Repair(models.Model):
                                            'result_package_id': False,
                                            'location_id': operation.location_id.id, #TODO: owner stuff
                                            'location_dest_id': operation.location_dest_id.id,})],
-                    'consume_repair_id': repair.id,
+                    'repair_id': repair.id,
                 })
                 moves |= move
                 operation.write({'move_id': move.id, 'state': 'done'})

--- a/None
+++ b/addons/mrp_repair/models/stock_traceability.py
@@ -0,0 +1,13 @@
+from odoo import models, api
+
+class MrpStockReport(models.TransientModel):
+    _inherit = 'stock.traceability.report'
+
+    @api.model
+    def get_links(self, move_line):
+        res_model, res_id, ref = super(MrpStockReport, self).get_links(move_line)
+        if move_line.move_id.repair_id:
+            res_model = 'mrp.repair'
+            res_id = move_line.move_id.repair_id.id
+            ref = move_line.move_id.repair_id.name
+        return res_model, res_id, ref
